bin/prado-cli.php

```diff
@@ -21,22 +22,21 @@
 	include($autoloader);
 }
 
-use Prado\Shell\TShellInterpreter;
-
-use Prado\Shell\Actions\TActiveRecordGenAction;
-use Prado\Shell\Actions\TActiveRecordGenAllAction;
-use Prado\Shell\Actions\TAppAction;
-use Prado\Shell\Actions\TFlushCachesAction;
-use Prado\Shell\Actions\TPhpShellAction;
-
 restore_exception_handler();
 
-//register action classes
-TShellInterpreter::getInstance()->addActionClass('Prado\\Shell\\Actions\\TAppAction');
-TShellInterpreter::getInstance()->addActionClass('Prado\\Shell\\Actions\\TFlushCachesAction');
-TShellInterpreter::getInstance()->addActionClass('Prado\\Shell\\Actions\\TPhpShellAction');
-TShellInterpreter::getInstance()->addActionClass('Prado\\Shell\\Actions\\TActiveRecordGenAction');
-TShellInterpreter::getInstance()->addActionClass('Prado\\Shell\\Actions\\TActiveRecordGenAllAction');
+$app_dir = dirname(__DIR__, 4);
```

Comment:
This seems to be the line causing prado-cli trying to load the application from /var in my error

Reply:
Do you have Prado in an application via composer?  It sounds like a machine specific installation outside composer.  The base directory being presumed is:
/directory_to_app/web_app/protected/vendor/prado-cli which links to
/directory_to_app/web_app/protected/vendor/prado/pradosoft/bin/

Hence shaving off 4 directories to get the app directory.  Outside the bin, -d option is for setting the app_dir.

Should this be checking another directory for being the app dir?  It sounds like it should, given your setup.

Reply:
After reviewing Yii and other platforms, They require an application configuration to run.

This PR changes the assumption that Prado doesn't need an application to run.  Before this PR, Prado was able to perform some base level functionality effectively outside the application framework.  The main functionalities of the shell required an application directory path.

This PR internalizes the application directory path based upon composer and automatically starts up the app config to provide the functionality.  All functionalities remove the requirement of providing an app directory path because it is built in.

This case covers most installations.  I saw your comment above with the error and it does seem that your Prado-master is operating outside the context of composer (and for development purposes, rightly so).

In such situation a "-d=/path/to/app_config" option is needed to load the application.  Removing the app config directory into an option standardizes the commands and creates the application context prior to running the interpreter/application.  This was also to remove the hack "app action" for application actions that needed a loaded application context.

I hope this assumption is not too far out there.   Yii has a similar implementation of their console.   This PR is modeled on the Yii command line so we can capture some people from that bankwagon and make it easy for ppl to grasp coming from that background.

Reply:
Should the error (no application in provided directory, try -d to specify your own directory) be more graceful?  That is what I am thinking.

Reply:
I think checking the current directory (and possible ./protected) for valid app_directory would also be a good thing to have as well

Reply:
I'm putting an app config check function in prado-cli.  It checks the composer vendor directory, and if the application configuration is not found (like in your instance), it checks the current working directory C.W.D., and if not found there, checks the CWD "/protected" directory.  This should cover your instance.

If not found in these three default directories, throws a proper Prado Shell Error telling the use to specify the app config directory with option "-d="

If the user specified the directory with "-d" but no application is found, checks the 'protected' sub-directory.  if not found throws a proper Prado Shell Error.

This is being tested and will check in soon.  In the mean time, you can use "-d=." to specify the current directory as the application to operate from.  The checkin will eliminate the need to specify the app directory in your instance (where the app directory is the CWD)

Reply:
>  It sounds like a machine specific installation outside composer.
You are right, sorry about this. I forgot i had some messy symlink structure to test prado master and PRs from the app.
Theoretically someone could cause the same problem installing prado globally using composer (composer global require), but I don't think that's a common (if even correct) usage at all.

> Should the error (no application in provided directory, try -d to specify your own directory) be more graceful? That is what I am thinking.

Yep, we should definitely provide an error message if the app directory is not found. The php backtrace for `Prado\TApplication->resolvePaths()` is just too cryptic.
Previously there used to be an "unable to load Prado application from directory xxx" error message

> I'm putting an app config check function in prado-cli. It checks the composer vendor directory, and if the application configuration is not found (like in your instance), it checks the current working directory C.W.D., and if not found there, checks the CWD "/protected" directory. This should cover your instance.

That would be awesome, thank you

Reply:
Fantastic.  I wasn't sure how to handle a Prado install from a repo but it's been figured out here.  I'll be posting the update soon.

Reply:
don't forget to try "-d=/tmp" to trigger an error so you can see what that looks like.